db/db.py

```
@@ -329,23 +354,45 @@ def read_all_profiles() -> list[BlueskyProfile]:
+        sqlite3.OperationalError: If database operation fails
     """
     with get_connection() as conn:
         rows = conn.execute("SELECT * FROM bluesky_profiles").fetchall()
         
-        return [
-            BlueskyProfile(
+        profiles = []
+        for row in rows:
+            # Validate required fields are not NULL
+            if row["handle"] is None:
+                raise ValueError("handle cannot be NULL")
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)